File: src/main/kotlin/com/asianpaints/apse/service_engineer/domain/entity/SiteAreaImages.java
package com.asianpaints.apse.service_engineer.domain.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Builder()
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "site_area_image")
public class SiteAreaImages {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
//    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.DETACH)
//    @JsonBackReference
//    @JoinColumn(name = "site_area_id", nullable = false)

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JsonBackReference
    @JoinColumn(name = "site_area_id", nullable = false)
    public SiteArea siteArea;
    @Column(name = "image_url", length = 500, nullable = false)
    public String imageUrl;
    @Column(name = "uploaded_at", nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime uploadedAt = LocalDateTime.now();
}

File: src/main/kotlin/com/asianpaints/apse/service_engineer/domain/entity/SiteAreaImages.kt
package com.asianpaints.apse.service_engineer.domain.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "site_area_image")
data class SiteAreaImages(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @ManyToOne(fetch = FetchType.EAGER, cascade = [CascadeType.DETACH])
    @JsonBackReference
    @JoinColumn(name = "site_area_id", nullable = false)

    val siteArea: SiteArea,

    @Column(name = "image_url", length = 500, nullable = false)
    val imageUrl: String,

    @Column(name = "uploaded_at", nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    val uploadedAt: LocalDateTime = LocalDateTime.now()
)
